packages/@lwc/engine/src/framework/base-lightning-element.ts

```diff
@@ -316,13 +314,6 @@ BaseLightningElementConstructor.prototype = {
                     )}: 1 argument required, but only 0 present.`
                 );
             }
-            if (!(event instanceof GlobalEvent)) {
```

Comment:
I think we need to still have a condition, just not the instanceof. The problem is that in the next line you're going to access `type` our of the event instance. Maybe a try/catch on that access is sufficient. Still this is subject to poisoning. I believe the regex condition below should be moved out into restrictions rather than here. Let's chat about it.